src/interactive.jl

```diff
@@ -0,0 +1,179 @@
+"""
+    generate([pkg::AbstractString]) -> Template
+
+Shortcut for `Template(; interactive=true)(pkg)`.
+If no package name is supplied, you will be prompted for one.
+"""
+function generate(pkg::AbstractString=prompt(Template, String, :pkg))
+    t = Template(; interactive=true)
+    t(pkg)
+    return t
+end
+
+"""
+    interactive(T::Type{<:Plugin}) -> T
+
+Interactively create a plugin of type `T`. Implement this method and ignore other
+related functions only if you want completely custom behaviour.
+"""
+function interactive(T::Type)
+    pairs = interactive_pairs(T)
+
+    # There must be at least 2 MultiSelectMenu options.
+    # If there are none, return immediately.
+    # If there's just one, add a "dummy" option.
+    isempty(pairs) && return T()
+    just_one = length(pairs) == 1
+    just_one && push!(pairs, :None => Nothing)
+
+    menu = MultiSelectMenu(
+        collect(map(pair -> string(first(pair)), pairs));
+        pagesize=length(pairs),
+    )
+    println("$(nameof(T)) keywords to customize:")
+    customize = sort!(collect(request(menu)))
+
+    # If the "None" option was selected, don't customize anything.
+    just_one && lastindex(pairs) in customize && return T()
+
+    kwargs = Dict{Symbol, Any}()
+    foreach(pairs[customize]) do (name, F)
+        kwargs[name] = prompt(T, F, name)
+    end
+    return T(; kwargs...)
+end
+
+struct NotCustomizable end
+
+"""
+    customizable(::Type{<:Plugin}) -> Vector{Pair{Symbol, DataType}}
+
+Return a list of keyword arguments that the given plugin type accepts,
+which are not fields of the type, and should be customizable in interactive mode.
+For example, for a constructor `Foo(; x::Bool)`, provide `[x => Bool]`.
+If `T` has fields which should not be customizable, use `NotCustomizable` as the type.
+"""
+customizable(::Type) = ()
+
+function pretty_message(s::AbstractString)
+    replacements = [
+        r"Array{(.*?),1}" => s"Vector{\1}",
+        r"Union{Nothing, (.*?)}" => s"Union{\1, Nothing}",
+    ]
+    return reduce((s, p) -> replace(s, p), replacements; init=s)
+end
+
+"""
+    input_tips(::Type{T}) -> Vector{String}
+
+Provide some extra tips to users on how to structure their input for the type `T`,
+for example if multiple delimited values are expected.
+"""
+input_tips(T::Type{<:Vector}) = ["comma-delimited", input_tips(eltype(T))...]
```

Comment:
I think this is slightly cleaner and faster
```suggestion
input_tips(::Type{Vector{T}}) where T = ["comma-delimited", input_tips(T)...]
```